akka-cluster/src/multi-jvm/scala/akka/cluster/NodeChurnSpec.scala

```diff
@@ -0,0 +1,122 @@
+import akka.remote.testkit.MultiNodeSpec
+import akka.testkit._
+import com.typesafe.config.ConfigFactory
+import org.scalatest.BeforeAndAfter
+import akka.actor.ActorSystem
+import akka.actor.ActorRef
+import akka.event.Logging.Info
+import akka.actor.Actor
+import akka.actor.Props
+
+object NodeChurnMultiJvmSpec extends MultiNodeConfig {
+  val first = role("first")
+  val second = role("second")
+  val third = role("third")
+
+  commonConfig(debugConfig(on = false).
+    withFallback(ConfigFactory.parseString("""
+      akka.cluster.auto-down-unreachable-after = 1s
+      akka.remote.log-frame-size-exceeding = 2000b
+      """)).
+    withFallback(MultiNodeClusterSpec.clusterConfig))
+
+  class LogListener(testActor: ActorRef) extends Actor {
+    def receive = {
+      case Info(_, _, msg: String) if msg.startsWith("New maximum payload size for [akka.cluster.GossipEnvelope]") ⇒
+        testActor ! msg
+      case _ ⇒
+    }
+  }
+}
+
+class NodeChurnMultiJvmNode1 extends NodeChurnSpec
+class NodeChurnMultiJvmNode2 extends NodeChurnSpec
+class NodeChurnMultiJvmNode3 extends NodeChurnSpec
+
+abstract class NodeChurnSpec
+  extends MultiNodeSpec(NodeChurnMultiJvmSpec)
+  with MultiNodeClusterSpec with ImplicitSender {
+
+  import NodeChurnMultiJvmSpec._
+
+  def seedNodes: immutable.IndexedSeq[Address] = Vector(first, second, third)
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+  }
+
+  val rounds = 3
+
+  override def expectedTestDuration: FiniteDuration = 45.seconds * rounds
+
+  def awaitAllMembersUp(additionaSystems: Vector[ActorSystem]): Unit = {
+    val numberOfMembers = roles.size + roles.size * additionaSystems.size
+    awaitMembersUp(numberOfMembers)
+    awaitAssert {
+      additionaSystems.foreach { s ⇒
+        val c = Cluster(s)
+        c.state.members.size should be(numberOfMembers)
+        c.state.members.forall(_.status == MemberStatus.Up)
+      }
+    }
+  }
+
+  def awaitRemoved(additionaSystems: Vector[ActorSystem]): Unit = {
+    awaitMembersUp(roles.size, timeout = 40.seconds)
+    awaitAssert {
+      additionaSystems.foreach { s ⇒
+        Cluster(s).isTerminated should be(true)
+      }
+    }
+  }
+
+  "Cluster with short lived members" must {
+    "setup stable nodes" taggedAs LongRunningTest in within(15.seconds) {
+      val logListener = system.actorOf(Props(classOf[LogListener], testActor), "logListener")
+      system.eventStream.subscribe(logListener, classOf[Info])
+      cluster.joinSeedNodes(seedNodes)
+      awaitMembersUp(roles.size)
+    }
+
+    "join and remove transient nodes without growing gossip payload" taggedAs LongRunningTest in {
+      // This test is configured with log-frame-size-exceeding and the LogListener
+      // will send to the testActor if unexpected increase in message payload size.
+      // It will fail after a while if vector clock entries of removed nodes are not pruned.
+      for (n ← 1 to rounds) {
+        log.info("round-" + n)
```

Comment:
it would, I did an [attempt](https://github.com/akka/akka/pull/1735) a long time ago, but we disagreed on the level of ambition and never got time to work on it again